lib/_element.js

```diff
@@ -344,6 +352,70 @@ function element () {
     }))
   }
 
+  Element.prototype.compareDocumentPosition = function (other) {
+    function getFirstNodeByOrder (nodes, nodeOne, nodeTwo) {
+      return nodes.reduce(function (result, node) {
+        return result !== false
+          ? result
+          : node === nodeOne
+            ? nodeOne
+            : node === nodeTwo
+              ? nodeTwo
+              : node.childNodes
+                ? getFirstNodeByOrder(node.childNodes, nodeOne, nodeTwo)
+                : false
+      }, false)
+    }
+
+    function isSourceContainTarget (source, target) {
+      while (target.parentNode) {
+        target = target.parentNode
+        if (target === source) {
+          return true
+        }
+      }
+      return false
+    }
+
+    function hasContaining (node, other) {
```

Comment:
I'd probably call this `eitherContains` and call the arguments `a` and `b` or `left` and `right`. Since the order of the arguments to this function doesn't matter. What do you think?